src/galaxy/SectorGenerator.cpp

```diff
@@ -109,12 +263,17 @@ const std::string SectorRandomSystemsGenerator::GenName(RefCountedPtr<Galaxy> ga
 
 	Uint32 weight = rng.Int32(chance);
 	if (weight < 500 || galaxy->GetFactions()->IsHomeSystem(SystemPath(sx, sy, sz, si))) {
-		/* well done. you get a real name  */
-		int len = rng.Int32(2, 3);
-		for (int i = 0; i < len; i++) {
-			name += sys_names[rng.Int32(0, SYS_NAME_FRAGS - 1)];
+		// well done. you get a "real" name
+		int nameGen = rng.Int32(0, 3);
+		switch (nameGen) {
+		case 0: FrontierNames::GetName(name, rng); break;
+		case 1: HybridNames::GetName(name, rng); break;
+		case 2: Doomdark::GetName(name, rng); break;
+		case 3: Katakana::GetName(name, rng); break;
+		default:
+			FrontierNames::GetName(name, rng);
+			break;
```

Comment:
Not something you have to worry about at all for this PR, but is this being called at a late enough point in sector generation that the assigned faction could bias the choice of name generator? I think that's something that would be nice to have support for in faction definitions.

Reply:
I did wonder about that, but this PR had lingered for a while already.